File: gblib/src/main/java/com/zwsi/gblib/GBSector.java
// Copyright 2018 Louis Perrochon. All rights reserved

// Deals with Sectors

package com.zwsi.gblib;

public class GBSector {
    int type = -1;        // nonexisting type
    String type_symbol = " ";
    int population = 0;   // nobody lives here (yet)
    GBRace owner = null;  // Controlling race. we only want object reference to the race, not a copy of the object.

    GBSector() {

    }

    String consoleDraw(GBData data) {
        if (population == 0) {
            return " " + type_symbol + " ";
        } else {
            //return " \u001B[7m" + GBData.sectorTypeToSymbol(type) + "\u001B[m ";
            //return " [" + type_symbol + "]";
            return " " + Integer.toString(owner.id) + " ";
        }
    }

}

File: gblib/src/main/java/com/zwsi/gblib/GBSector.kt
// Copyright 2018-2019 Louis Perrochon. All rights reserved

// Deals with Sectors

package com.zwsi.gblib

import com.squareup.moshi.JsonClass
import com.zwsi.gblib.GBController.Companion.u

@JsonClass(generateAdapter = true)
data class GBSector constructor(val uidPlanet: Int) {

    // Fixed Properties: type, typeSymbol, revenue
    var type = -1        // nonexisting type
    internal var typeSymbol = "?"
    internal var revenue = 0
    var maxPopulation = 0

    var population = 0
    var uidSectorOwner: Int = -1
    val sectorOwner: GBRace
        get() = u.race(uidSectorOwner)

    // TODO: This should be part of the constructor
    fun chooseType(planetIdxType: Int) {
        type = GBData.sectorTypesChance[planetIdxType][GBData.rand.nextInt(10)]
        typeSymbol = GBData.sectorTypeConsoleFromIdx(type)
        revenue = GBData.sectorMoneyFromIdx(type)
        maxPopulation = GBData.sectorMaxPopulationFromIdx(type)
    }

    internal fun consoleDraw(): String {
        if (population == 0) {
            return " $typeSymbol "
        } else {
            //return " \u001B[7m $type_symbol \u001B[m ";
            return " $uidSectorOwner "
            //return "[$type_symbol]"

        }
    }
}

